asapdiscovery-ml/asapdiscovery/ml/weights/weights.py

```diff
@@ -47,114 +45,50 @@ def fetch_model_from_spec(
+        weights_hash = weights["sha256hash"]
 
-def fetch_file(
-    filename: str, file_dir: str = "./_weights/", force_fetch: bool = False
-) -> Path:
-    """Fetch weights from remote and save to local path.
-
-    Parameters
-    ----------
-    filename : str
-        Remote url or local path to file.
-    file_dir : str, default="./_weights/"
-        Local path to save weights if a remote url is provided, or to check if weights exist locally, by default "./_weights/"
-    force_fetch : bool, default=False
-        Force fetch file from remote, by default False
-
-    Returns
-    -------
-    Path to file.
-    """
-
-    is_url = validators.url(filename)
-
-    if not file_dir:
-        raise ValueError("file_dir must be provided and must not be falsy.")
+        registry = {}
+        registry[weights_resource] = f"sha256:{weights_hash}"
 
-    if is_url:
-        logging.info("file is a remote url")
-        logging.info(f"Fetching file from {filename}")
-        if force_fetch:
-            logging.info("Force fetching file from remote")
-            # fetch from remote
-            target_file = download_file(filename, file_dir)
-            return target_file
+        # fetch config if provided
+        if "config" in model_spec:
+            config = model_spec["config"]
+            config_resource = config["resource"]
+            config_hash = config["sha256hash"]
+            registry[config_resource] = f"sha256:{config_hash}"
         else:
-            # check if file exists locally
-            local_filename = filename.split("/")[-1]
-            local_path = Path(file_dir).joinpath(Path(local_filename))
-            if local_path.exists():
-                # exists locally, skip fetch from remote
-                logging.info(
-                    f"filename exist locally at {local_path}, skipping fetch from remote"
-                )
-                return Path(local_path)
-            else:
-                # fetch from remote
-                logging.info("filename does not exist locally, fetching from remote")
-                target_file = download_file(filename, file_dir)
-                return target_file
-    else:
-        local_path = Path(file_dir).joinpath(Path(filename))
-
-        logging.info(f"Fetching file from {local_path}")
-        if not Path(local_path).exists():
-            raise FileNotFoundError(
-                f"Local file {local_path} does not exist, provide a valid remote url or local path to file"
-            )
-
+            config_resource = None
+
+        # make pooch registry
+        subdir = model
+        registry = pooch.create(
+            path=Path(local_dir).joinpath(Path(subdir)),
+            base_url=base_url,
+            registry=registry,
+        )
+
+        # fetch weights
+        weights_file = Path(registry.fetch(weights_resource))
+        # fetch config
+        if config_resource:
+            config_file = Path(registry.fetch(config_resource))
```

Comment:
it looks `Pooch.fetch` can raise exceptions, so we should handle those in case of failure